server/src/main/java/io/crate/types/DateType.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed to CRATE Technology GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.types;
+
+import io.crate.Streamer;
+import org.apache.lucene.util.RamUsageEstimator;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.ZoneOffset;
+
+public class DateType extends DataType<Long>
+    implements FixedWidthType, Streamer<Long> {
+
+    public static final int ID = 23;
+    public static final DateType INSTANCE = new DateType();
+    public static final int DATE_SIZE = (int) RamUsageEstimator.shallowSizeOfInstance(Long.class);
+
+    @Override
+    public int id() {
+        return ID;
+    }
+
+    @Override
+    public Precedence precedence() {
+        return Precedence.DATE;
+    }
+
+    @Override
+    public String getName() {
+        return "date";
+    }
+
+    @Override
+    public Streamer<Long> streamer() {
+        return this;
+    }
+
+    @Override
+    public Long implicitCast(Object value) throws IllegalArgumentException, ClassCastException {
+        if (value == null) {
+            return null;
+        } else if (value instanceof java.util.Date) {
+            return ((java.util.Date) value).getTime();
+        }  else if (value instanceof java.sql.Date) {
```

Comment:
The value can never be of type `java.sql.Date` or `java.util.Date` so these types must not be handled.

Reply:
Done